### PR TITLE
Add gpu selector to Avalonia

### PR DESCRIPTION
**Pull request by emmauss**

### CODE REVIEW

File: Ryujinx.Ava/Ui/Backend/Vulkan/VulkanPhysicalDevice.cs [hunk 1]
```diff
@@ -50,55 +55,66 @@ private VulkanPhysicalDevice(PhysicalDevice apiHandle, Vk api, uint queueCount,
...
 
             foreach (var physicalDevice in physicalDevices)
             {
                 instance.Api.GetPhysicalDeviceProperties(physicalDevice, out var properties);
-                physicalDeviceProperties.Add(physicalDevice, properties);
+                PhysicalDevices.Add(physicalDevice, properties);
             }
 
+            SuitableDevices = PhysicalDevices.Where(x => IsSuitableDevice(instance.Api,
+                        x.Key,
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/Backend/Vulkan/VulkanPhysicalDevice.cs:67]: Adjust indentation.

File: Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs [hunk 2]
```diff
@@ -355,6 +419,20 @@ public void SaveSettings()
             config.System.Language.Value = (Language)Language;
             config.System.Region.Value = (Region)Region;
 
+            var selectedGpu = _gpuIds.ElementAtOrDefault(PreferredGpuIndex);
+            if(!requiresRestart)
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs:423]:
```suggestion
            if (!requiresRestart)
```

File: Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs [hunk 3]
```diff
@@ -355,6 +419,20 @@ public void SaveSettings()
...
+            {
+                var platform = AvaloniaLocator.Current.GetService<VulkanPlatformInterface>();
+                if(platform!= null)
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs:426]:
```suggestion
                if (platform!= null)
```

File: Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs [hunk 4]
```diff
@@ -393,6 +471,21 @@ public void SaveSettings()
...
+
+            if (requiresRestart)
+            {
+                var choice = await ContentDialogHelper.CreateChoiceDialog(
+                    _owner,
+                    LocaleManager.Instance["SettingsAppRequiredRestartMessage"],
+                    LocaleManager.Instance["SettingsGpuBackendRestartMessage"],
+                    LocaleManager.Instance["SettingsGpuBackendRestartSubMessage"]);
+
+                if(choice)
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/ViewModels/SettingsViewModel.cs:483]:
```suggestion
                if (choice)
```

**Comment by emmauss** [comment]
Addressed

**Review by gdkchan — APPROVED** [review]
Thanks.

**Pull request merged**